File: interface/src/main/java/com/kgc/pinyougou/service/ItemCatService.java
package com.kgc.pinyougou.service;

import com.kgc.pinyougou.domain.ItemCat;
import com.kgc.pinyougou.utils.PageResult;

import java.util.List;

/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface ItemCatService {

	/**
	 * @Author Admin
	 * @description:  //TODO
	 * @param:
	 * @param pageNum   分页页数
	 * @param pageSize  分页记录数
	 * @return: com.kgc.pinyougou.utils.PageResult
	 * @time: 2020/4/25 22:11
	 */
	PageResult findAllItemCat(Integer pageNum, Integer pageSize, ItemCat itemCat);

	List<ItemCat> findByParentId(Long id);

	Integer insertItemCat(ItemCat itemCat);

	ItemCat findOne(Long id);

	Integer updateItemCat(ItemCat itemCat);

	Integer deleteItemCat(Long [] ids);
}

File: pinyougou-manager-web/src/main/resources/static/js/service/itemCatService.js
//服务层
app.service('itemCatService',function($http){

    this.findAllItemCat=function (entity) {
        return $http.post("../itemCat/findAllItemCat,entity")
    }

    this.findByParentId=function (id) {
        return $http.get("../itemCat/findByParentId?parentId="+id)
    }

    this.findOne=function (id) {
        return $http.get("../itemCat/findOne?id="+id)
    }

    this.del=function (ids) {
        return $http.post("../itemCat/delItemCat",ids)
    }

    this.add=function (entity) {
        return $http.post("../itemCat/addItemCat",entity)
    }

    this.update=function (entity) {
        return $http.post("../itemCat/updateItemCat",entity)
    }
});
